lib/straight-server/bip70/payment_request.rb

```diff
@@ -0,0 +1,84 @@
+require_relative 'paymentrequest.pb'
+
+module StraightServer
+
+  module Bip70
+
+    class PaymentRequest
+
+      def initialize(order:)
+        @order = order
+
+        output   = create_output
+        details  = create_payment_details(output)
+
+        @payment_request = Payments::PaymentRequest.new
+        @payment_request.payment_details_version    = 1
+        @payment_request.serialized_payment_details = details.to_s
+
+        @payment_request.pki_type, @payment_request.pki_data = create_pk_infrastructure
+
+        @payment_request.signature = ''
+        @payment_request.signature = create_signature(@payment_request.to_s)
+      end
+
+      def to_s
+        @payment_request.to_s
+      end
+
+      private
+
+        def create_output
+          output = Payments::Output.new
+          output.amount = @order.amount
+          output.script = @order.address
+          output
+        end
+
+        def create_payment_details(output)
+          payment_details = Payments::PaymentDetails.new
+          payment_details.network = @order.gateway.test_mode ? 'test' : 'main'
+          payment_details.time = @order.created_at.to_i
+          payment_details.expires = @order.created_at.to_i + @order.gateway.orders_expiration_period
+          payment_details.memo = 'Payment request for GearPoweredMerchant'
+          payment_details.payment_url = ''
+          payment_details.merchant_data = ''
+          payment_details.outputs << output
+          payment_details
+        end
+
+        def create_pk_infrastructure
+          if Config.ssl_certificate_path
+            pki_data = create_pki_data
+            ['x509+sha256', pki_data.to_s]
+          else
+            ['none', '']
+          end
+        end
+
+        def create_pki_data
+          pki_data = Payments::X509Certificates.new
+
+          certificates = File.read(Config.ssl_certificate_path)
+          certificates.each_line("-----END CERTIFICATE-----\n") do |cert|
+            pki_data.certificate << OpenSSL::X509::Certificate.new(cert).to_der
+          end
+
+          pki_data
+        end
+
+        def create_signature(data)
+          key =
+            if Config.private_key_path
+              File.read(Config.private_key_path)
+            else
+              2048
```

Comment:
In this case, it will generate a new private key for each signature?
I guess, it's better just to raise exception if config is empty.